lua/orgmode/objects/calendar.lua

```diff
@@ -149,69 +161,177 @@ function Calendar.render()
   else
     table.insert(content, ' [i] - enter date')
   end
+  if Calendar.has_time() then
```

Comment:
Would be great to have a mapping that exists "time" mode and goes back to selecting a date.

Reply:
I implemented it in the way, that you can exit the time with Esc and with another Esc exit the calendar completely and this feels quite naturally for me. Adding another mapping explicit for the time sounds cumbersome to me. Maybe Tab to switch between calendar and time but actually I currently don't see so much value to change what already works quite good.

Reply:
Because I rebased my code onto your refactoring, the window now also closes with `q`. I mapped the switch back to day selection now to `d`.